### PR TITLE
Refactor & Improve Route53

### PR DESCRIPTION
**Pull request by Eelviny**
### Step 1: Link to Jira issue
https://stackstate.atlassian.net/browse/STAC-13208

### Step 2: Description of changes
- Refactor route53_hostedzone and route53_domains
- Add create_arn functions for route53 and route53domains
- Fetch more domain information
- Add Schematics on API calls
- Add IAM permission checking
- Make route53 hosted zone name more user-friendly

### Step 3: Did you add / update tests for your changes in the right area?
- [x] Unit/Component test
- [ ] Integration test	

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

**Comment by Eelviny** [comment]
Blocked by #152

File: aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py
Comment (Jacco) [inline aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:28]: Think this one should be required

File: aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py
Comment (Jacco) [inline aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:24]: Required

File: aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py
Comment (Jacco) [inline aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:48]: Tags collections often do support paginate. We need to check this for all APIs. Often they have a "HasMoreTags" attribute. Don't know if that supports the can_paginate call.

**Review by Jacco — CHANGES_REQUESTED** [review]

File: aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py
Comment (Eelviny) [reply aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:48]: This particular resource [does not](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/route53.html#Route53.Client.list_tags_for_resource) - let's worry about this with a pass over all resources later

**Comment by Eelviny** [comment]
This is failing due to the tests needing to be fixed, but will wait until #152 is merged

File: aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py
Comment (Jacco) [inline aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py:75]: Here we do not use the CLOUDFORMATION_TYPE (we want to get rid of)

File: aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py
Comment (Jacco) [inline aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py:77]: We need to make sure that a CF-template resources that is a route53 domain gets references correctly. (just verify route)

**Review by Jacco — CHANGES_REQUESTED** [review]

File: aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py
Comment (Eelviny) [reply aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py:77]: Actually route53 domains is not supported by Cfn... I kinda made up the Cfn type, according to how it would be

**Review by Jacco — APPROVED** [review]

File: aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py
Comment (zandrewitte) [inline aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py:41]: I think the `deepcopy` is unnecessary, we're not sharing this dict right?

File: aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py
Comment (zandrewitte) [inline aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:96]: Can this potentially break?

File: aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py
Comment (zandrewitte) [inline aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:20]: Do we really need this default? If we can't get the name can we uniquely identify this hosted zone?

Comment (zandrewitte) [reply aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:20]: I see we use the hosted zone id as the external id, maybe we can fallback to that instead of unknown for the name

Comment (zandrewitte) [reply aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:20]: You will probably have to keep this here then and add a check in `process_hosted_zone`

**Review by zandrewitte — CHANGES_REQUESTED** [review]

File: aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py
Comment (Eelviny) [reply aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:96]: Unlikely - if a value is not found, Schematics has a default value of `UNKNOWN`. If no forward slash is found, then it will return the whole string as usual

File: aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py
Comment (Eelviny) [reply aws_topology/stackstate_checks/aws_topology/resources/route53_domain.py:41]: In general I've been taught that you should not mutate a reference that is passed into a function, but it won't break anything if I don't do it.

I don't mind either way, let me know and I'll remove it

File: aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py
Comment (Eelviny) [reply aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py:96]: The reason for this is the format is `/hostedzone/<id of resource>` - this just removes the first section

**Review by zandrewitte — APPROVED** [review]

**Pull request merged**
